client/src/tokens/typeScale.js

```diff
@@ -0,0 +1,80 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const plugin = require('tailwindcss/plugin');
+
+// TypeScale plugin:
+// This plugin generates component classes using tailwind's configuration for each object inside of the typeScale const.
+// If the tailwind config is using a prefix such as 'w-' this will be included in the compiled css class eg. .w-h1
+module.exports = plugin(({ addComponents, theme }) => {
+  const headingBaseStyles = {
+    fontWeight: theme('fontWeight.bold'),
+    color: theme('colors.primary.DEFAULT'),
+    lineHeight: theme('lineHeight.tight'),
+  };
+
+  const typeScale = {
+    '.h1': {
+      fontSize: theme('fontSize.30'),
+      fontWeight: theme('fontWeight.extrabold'),
+      color: theme('colors.primary.DEFAULT'),
+      lineHeight: theme('lineHeight.tight'),
+    },
+    '.h2': {
+      fontSize: theme('fontSize.24'),
+      ...headingBaseStyles,
+    },
+    '.h3': {
+      fontSize: theme('fontSize.22'),
+      ...headingBaseStyles,
+    },
+    '.h4': {
+      fontSize: theme('fontSize.18'),
+      ...headingBaseStyles,
+    },
+    '.h5': {
+      fontSize: theme('fontSize.16'),
+      ...headingBaseStyles,
+    },
+    '.h6': {
+      fontSize: theme('fontSize.15'),
+      ...headingBaseStyles,
+    },
+    '.label-1': {
```

Comment:
I think my old man eyes prefer the dashed approach `label-1` just for a visual separation. If it helps this is consistent with all the other class names that use numbers this way eg: `border-1` `color-type-100`

Reply:
But I guess by that logic, h1,h2,h3, should be dashed then too 😅 but since this is a common name I figured they'd make more sense as is for headings. So now i'm not sure either

Reply:
We could use t-shirts for the label sizing as well. `label-sm` etc